File: src/main/java/org/goafabric/invoice/process/adapter/invoice/InvoiceMockAdapter.java
package org.goafabric.invoice.process.adapter.invoice;

import org.springframework.aot.hint.annotation.RegisterReflectionForBinding;
import org.springframework.stereotype.Component;

import java.nio.charset.StandardCharsets;
import java.util.Base64;

@Component
@RegisterReflectionForBinding(Invoice.class)
public class InvoiceMockAdapter  {

    public void check(Invoice invoice) {
    }

    public Invoice encrypt(Invoice invoice) {
        var encryptedContent = Base64.getEncoder().encodeToString(
                invoice.content().getBytes(StandardCharsets.UTF_8));
        return new Invoice(invoice.id(), encryptedContent);
    }

    public void send(Invoice invoice) {
        // just empty
    }

    public Invoice store(Invoice invoice) {
        return invoice;
    }
}

File: src/main/kotlin/org/goafabric/invoice/process/adapter/invoice/InvoiceMockAdapter.kt
package org.goafabric.invoice.process.adapter.invoice

import org.goafabric.invoice.process.adapter.invoice.dto.Invoice
import org.springframework.stereotype.Component
import java.nio.charset.StandardCharsets
import java.util.*

@Component
class InvoiceMockAdapter {
    fun create(): Invoice {
        return Invoice(UUID.randomUUID().toString(), "Example Content")
    }

    fun check(invoice: Invoice) {
        check(!invoice.content.isEmpty()) { "Validation failed" }
    }

    fun encrypt(invoice: Invoice): Invoice {
        val encryptedContent = Base64.getEncoder().encodeToString(
            invoice.content.toByteArray(StandardCharsets.UTF_8)
        )
        return Invoice(invoice.id, encryptedContent)
    }

    fun send(invoice: Invoice?) {
    }

    fun store(invoice: Invoice?) {
    }
}
